ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt

```diff
@@ -98,18 +106,43 @@ private fun Template2MainContent(state: PaywallViewState.Loaded, viewModel: Payw
 private fun IconImage(uri: Uri?) {
     uri?.let {
         Column(modifier = Modifier.widthIn(max = Template2UIConstants.maxIconWidth)) {
-            RemoteImage(
-                urlString = uri.toString(),
-                modifier = Modifier
-                    .aspectRatio(ratio = 1f)
-                    .widthIn(max = Template2UIConstants.maxIconWidth)
-                    .clip(RoundedCornerShape(Template2UIConstants.iconCornerRadius)),
-                contentScale = ContentScale.Crop,
-            )
+            val modifier = Modifier
+                .aspectRatio(ratio = 1f)
+                .widthIn(max = Template2UIConstants.maxIconWidth)
+                .clip(RoundedCornerShape(Template2UIConstants.iconCornerRadius))
+            if (uri.toString().contains(PaywallData.defaultAppIconPlaceholder)) {
+                AppIcon(
+                    modifier = modifier,
+                )
+            } else {
+                RemoteImage(
+                    urlString = uri.toString(),
+                    modifier = modifier,
+                    contentScale = ContentScale.Crop,
+                )
+            }
         }
     }
 }
 
+@Composable
+private fun AppIcon(
+    modifier: Modifier = Modifier,
+) {
+    val context = LocalContext.current
+    val appIconResId = remember(context) {
```

Comment:
Hmm the app icon shouldn't really change, so I don't think we need to make it depend on the context? So this:
```suggestion
    val appIconResId = remember {
```